devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,209 @@
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	def endpoint = cmd.sourceEndPoint
+	if (endpoint == 1) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		String childDni = "${device.deviceNetworkId}:$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+	}
+}
+
+def on() {
+	resetActiveSound()
+	state.lastTriggeredSound = 1
+	response(zwave.basicV1.basicSet(value: 0xFF).format())
+}
+
+def off() {
+	response(zwave.basicV1.basicSet(value: 0x00).format())
+}
+
+def strobe() {
+	on()
+}
+
+def siren() {
+	on()
+}
+
+def both() {
+	on()
+}
+
+def ping() {
+	zwave.basicV1.basicGet()
+}
+
+private addChildren(numberOfSounds) {
+	for (def endpoint : 2..numberOfSounds) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = device.displayName[0..-2] + "$endpoint"
+
+			addChildDevice("Aeotec Child Siren", childDni, device.getHub().getId(), [
+					completedSetup: true,
+					label         : componentLabel,
+					isComponent   : true,
+					componentName : "Aeotec Doorbell 6 $endpoint",
+					componentLabel: "Aeotec Doorbell 6 $endpoint"
+			])
+		} catch (Exception e) {
+			log.debug "Excep: ${e} "
+		}
+	}
+}
+
+private encap(cmd, endpoint = null) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+
+def sendCommand(deviceDni, commands) {
+	def result = commands.collect {
+		if (it instanceof String) {
+			it
+		} else {
+			encap(it, channelNumber(deviceDni))
+		}
+	}
+	sendHubCommand(result, 100)
+}
+
+def channelNumber(String dni) {
+	dni?.split(":")[-1] as Integer
+}
+
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	if (cmd.notificationStatus == 0xFF) {
+		switch (cmd.event) {
+			case 0x09: //TAMPER
+				sendEvent([name: "tamper", value: "detected"])
+				break
+			case 0x01: //ON
+					if(state.lastTriggeredSound == 1 ) {
+						sendEvent([name: "alarm", value: "both"])
+					} else {
+						String childDni = "${device.deviceNetworkId}:$state.lastTriggeredSound"
+						def child = childDevices.find { it.deviceNetworkId == childDni }
+						child?.sendEvent([name: "alarm", value: "both"])
+					}
+				break
+			case 0x00: //OFF
+					resetActiveSound()
+					sendEvent([name: "tamper", value: "clear"])
+				break
+		}
+	}
+}
+
+def setOnChild(deviceDni) {
+	resetActiveSound()
+	sendCommand(deviceDni, zwave.basicV1.basicSet(value: 0xFF))
+	state.lastTriggeredSound = deviceDni?.split(":")[-1] as Integer
+	setActiveSound(state.lastTriggeredSound)
+}
+
+def setOffChild(deviceDni) {
+	sendCommand(deviceDni, zwave.basicV1.basicSet(value: 0x00))
+}
+
+def resetActiveSound() {
+	if(state.lastTriggeredSound == 1 ) {
+		sendEvent([name: "alarm", value: "off"])
+	} else {
+		String childDni = "${device.deviceNetworkId}:$state.lastTriggeredSound"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
```

Comment:
With as much as this code is used, you could probably just create a helper function that would take in a sound/child ID and maybe even an event map and the helper would find the child and send the event. Maybe call it `sendChildEvent` so it is more clear what it is doing (`setActiveSound` sounds like it would be causing the device to actually set the sound, instead of just updating the attribute as it does now).